include/openthread/platform/crypto.h

```diff
@@ -118,6 +120,14 @@ typedef struct otCryptoKey
     uint32_t       mKeyRef;    ///< The PSA key ref (requires `mKey` to be NULL).
 } otCryptoKey;
 
+typedef struct otCryptoContext
+{
+    void *   mContext;     ///< Pointer to the context.
+    uint16_t mContextSize; ///< The length of the context in bytes.
+    uint64_t mInPlaceStorage[OT_CRYPTO_CONTEXT_IN_PLACE_STORAGE_SIZE /
```

Comment:
I'm not sure if defining the maximum context size in OT header will always work.

For example, some platforms may implement HW acceleration and have custom contexts sizes e.g. for SHA256 like in nRF528xx: https://github.com/openthread/ot-nrf528xx/blob/main/third_party/NordicSemiconductor/libraries/nrf_security/include/mbedtls/sha256_alt.h#L30

So I would say either we need to have a way of overwriting it by the platform or figure out some different approach.

Reply:
+1 on @LuDuda  comment. 

I also don't think having a fixed storage size for all crypto APIs is good idea. It adds to stack use (e.g. while `mbedtls_aes_context` may require about 288 bytes, `mbedtls_sha256_context` requires much less and using same value for all may not be good).

Reply:
The new commit has made the local storage size configurable and different for each class. Please take a look.